var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -69,6 +70,11 @@ class Hiop(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('magma@2.5.4:', when='@0.4:+rocm')
     depends_on('magma@2.6.1:', when='@0.4.6:+rocm')
 
+    # MAGMA v2.6.2 was not released when v0.5.4 was released, but we need
+    # commits from master branch. The most recent tested commit is
+    # 5959b8783e45f1809812ed96ae762f38ee701972
+    depends_on('magma@master', when='@0.5.4:')
```

Comment:
I see that maintainer approved, but I wonder if we should have a tagged version depend on `magma@master` (which I assume is a moving target). @adamjstewart Do you have opinions on this?

Reply:
This is very much a concern for us as well, but Magma 2.6.2 was supposed to be released last November and we've needed this release for some time now. Another option might be to add a deprecated version to magma that points to the commit I mention in the comment just for this hiop (depreciated so other users won't accidentally use it).

Reply:
Would it be fine to:
1. Mark `magma@2.6.1` as preferred
2. Add a placeholder for the new magma release, e.g. `2.6.2rc1`, and pin it to that commit (without saying it's deprecated, which will make Spack trying actively to avoid it)

? In this way `hiop` will depend on a fixed commit rather than a moving branch.

Reply:
That seems like a great solution! With the sign off of Magma folks I'm happy to make this change. Does it belong in this PR?

Reply:
> Does it belong in this PR?

Yes, I would do it in this PR since it is required to have the new version of `hiop`.

Reply:
Yes, I would add a 2.6.2rc1 version to magma, make 2.6.1 preferred, and have hiop require `magma@2.6.2:`.

Reply:
On the magma site, we continued adding functionalities and didn't release 2.6.2 as planned. We can mark current master as 2.6.2 by the end of this week to resolve problems like this.